Client/Ecosia/Experiments/Unleash/BingDistributionExperiment.swift

```diff
@@ -0,0 +1,55 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Core
+
+struct BingDistributionExperiment {
+    
+    private init() {}
+    
+    static func incrementCounter() {
+        User.shared.searchCount += 1
+    }
+    
+    static var isEnabled: Bool {
+        Unleash.isEnabled(.bingDistribution)
+    }
+    
+    static private var isTestVariant: Bool {
+        return Unleash.getVariant(.bingDistribution).name == "test"
+    }
+    
+    static func bingSearchWithQuery(_ query: String) -> URL? {
+        guard let rootUrl = URL(string: "https://www.bing.com"),
+              var components = URLComponents(url: rootUrl, resolvingAgainstBaseURL: false) else {
+            return nil
+        }
+        components.path = "/search"
+        components.queryItems = [
+            URLQueryItem(name: "q", value: query),
+            URLQueryItem(name: "PC", value: "ECAA"),
+            URLQueryItem(name: "FORM", value: "ECAA01"),
+            URLQueryItem(name: "PTAG", value: "st_ios_bing_distribution_test"),
+        ]
+        return components.url
+    }
+    
+    static func ecosiaSearchWithTypetag(_ query: String) -> URL? {
+        let url = URL.ecosiaSearchWithQuery(query)
+        guard var components = URLComponents(url: url, resolvingAgainstBaseURL: false) else { return nil }
+        components.queryItems?.append(URLQueryItem(name: "tts", value: "st_ios_bing_distribution_control"))
+        return components.url
+    }
+    
+    static func searchURLForQuery(_ query: String) -> URL? {
+        if isTestVariant {
+            // Increment counter everytime we use bing's url
+            BingDistributionExperiment.incrementCounter()
```

Comment:
This is stateful behaviour for an apparently stateless method who is just supposed to return an URL. I would either suggest to make that increment explicit in the method name, or extract it to the caller.

Reply:
That's a good point, the caller is also a getter which in turn is called in multiple places and I was trying to avoid duplicated code. Anyway, I'll check how it looks if this is duplicated or make this stateful behaviour explicit 🙂

Reply:
Done, found it better to extract it to `BrowserViewController` and `SearchViewController` even if duplicated so that the state change is explicit 🙏